ui5/browser/controller/Browser.controller.js

```diff
@@ -741,11 +743,28 @@ sap.ui.define(['sap/ui/core/Component',
       },
 
       /** @brief Add Tab event handler */
-      addNewButtonPressHandler: function(oEvent) {
-         if (this.isConnected)
-            this.websocket.Send("NEWCANVAS");
+      addNewButtonPressHandler: async function(oEvent) {
+        var oButton = oEvent.getSource().mAggregations._tabStrip.mAggregations.addButton;
```

Comment:
@Falcort 
Thibault, please use correspondent ui5 methods to access event parameters.
Otherwise we will face lot of problem when ui5 internals will change.
I saw similar peace of code in other places

Reply:
I have done this because i didn't fine any other options on the SAP UI 5 page to target the new button
In other cases too

Reply:
Then you need to assign some unique Id for button and use such id as event parameters.
This is not a solution, but dirty workaround, which most probably will not work in the future.

Reply:
I am 100% ok about that, but this button is created with the line 
```
<TabContainer id="myTabContainer"
                      showAddNewButton="true"
                      addNewButtonPress="addNewButtonPressHandler"
                      itemSelect="tabSelectItem"
                      itemClose="tabCloseHandler">
```
```
showAddNewButton="true"
```
So you have no possibility to add a special ID to that button

I am also encountering the same problem for an other thing, if i use SAP to get the ID of the selected tab it return me __item30, but it doesn't exist, the correct one is __item31
If you have a solution to all those problems i will take if an refactor all my past code to apply it, but for now i don't know how to do differently

Reply:
As I mention, either use extra parameters when assign event handlers.
Or try to assign custom id to button your pressing.
There is definitely the solution.

Reply:
There is of course, the solution to create everything a string and a number, but to apply that i need to refactor almost everything in the Browser in order to make it usable.
I will take some times, and if you want right after the tabs selections i can start applying my solution, but i might take a week ?
An of course it will resolve all or almost all of those problems and should make the code easier to read

Reply:
>  i can start applying my solution, but i might take a week ?

Let me know when functionality with "+" sign and image viewer is there that I can merge it.
And then next task will be to avoid such constructs.

Reply:
Sure,
I already thought about that problems, and for me the solution will be to create each tabs manually, creating them with the id : tab + i
And i will be an auto increment number in the code
All sub-elements of that tab will have a text + i as ID, so i can easily target them with there ID, it will solve like almost every issue that i have encountered, and created such workaround
t will work for the tabContainer, but i need to find a solution for the + and the TreeTable